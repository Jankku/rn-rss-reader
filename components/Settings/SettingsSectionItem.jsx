import { useTheme } from '@react-navigation/native';
import { Pressable, StyleSheet } from 'react-native';

function SettingsSectionItem({ onPress, children }) {
  const { colors } = useTheme();
  const styles = makeStyles(colors);

  return (
    <Pressable onPress={onPress} style={styles.container}>
      {children}
    </Pressable>
  );
}

const makeStyles = (colors) =>
  StyleSheet.create({
    container: {
      backgroundColor: colors.card,
      padding: 16,
    },
  });

export default SettingsSectionItem;
