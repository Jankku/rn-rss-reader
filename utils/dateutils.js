import { DateTime } from 'luxon';

/**
 * For articles released today return time and date otherwise.
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date
 */
export function RFC2822ToTimeOrDate(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate =
    dateObject.toRelativeCalendar() === 'today'
      ? dateObject.toLocaleString(DateTime.TIME_24_SIMPLE)
      : dateObject.toLocaleString(DateTime.DATE_SHORT);
  return formattedDate;
}

/**
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date in DateTime format
 */
export function RFC2822ToDateTime(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate = dateObject.toLocaleString(DateTime.DATETIME_SHORT);
  return formattedDate;
}
