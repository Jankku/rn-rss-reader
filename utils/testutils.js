import { render } from '@testing-library/react-native';
import { ThemeContextProvider } from '../context/ThemeContext';
import { LocationContextProvider } from '../context/LocationContext';
import { RegionContextProvider } from '../context/RegionContext';

const AllProviders = ({ children }) => {
  return (
    <LocationContextProvider>
      <RegionContextProvider>
        <ThemeContextProvider>{children}</ThemeContextProvider>
      </RegionContextProvider>
    </LocationContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
