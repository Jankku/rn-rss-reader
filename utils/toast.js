import Toast from 'react-native-root-toast';

const showToast = (message) =>
  Toast.show(message, {
    position: 100,
    opacity: 0.9,
    shadow: false,
  });

export default showToast;
