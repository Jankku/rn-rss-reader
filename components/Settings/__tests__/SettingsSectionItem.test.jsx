import { Text } from 'react-native';
import SettingsSectionItem from '../SettingsSectionItem';
import { render, fireEvent } from '../../../utils/testutils';

describe('Test SettingsSectionItem', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(
      <SettingsSectionItem onPress={() => {}}>
        <Text>Theme</Text>
      </SettingsSectionItem>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should call onPress', () => {
    const onPress = jest.fn();
    const { getByText } = render(
      <SettingsSectionItem onPress={onPress}>
        <Text>Theme</Text>
      </SettingsSectionItem>
    );
    const item = getByText('Theme');

    fireEvent.press(item);
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
