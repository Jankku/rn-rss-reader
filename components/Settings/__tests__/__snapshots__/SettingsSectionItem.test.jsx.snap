// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SettingsSectionItem Should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(255, 255, 255)",
      "padding": 16,
    }
  }
>
  <Text>
    Theme
  </Text>
</View>
`;
